zarafa/__init__.py

```diff
@@ -1297,7 +1297,7 @@ def check_folder(folder):
     def item(self, entryid):
         """ Return :class:`Item` with given entryid; raise exception of not found """ # XXX better exception?
 
-        item = Item() # XXX copy-pasting..
+        item = Item(parent=self) # XXX copy-pasting..
```

Comment:
'self' is a Store, so not really the direct parent of the item.. but note that 'Item.folder' looks up the correct parent folder using PR_PARENT_ENTRYID.